rclcpp/src/rclcpp/clock.cpp

```diff
@@ -118,18 +137,30 @@ Clock::create_jump_callback(
     throw std::bad_alloc{};
   }
 
-  // Try to add the jump callback to the clock
-  rcl_ret_t ret = rcl_clock_add_jump_callback(
-    &rcl_clock_, threshold, Clock::on_time_jump,
-    handler.get());
-  if (RCL_RET_OK != ret) {
-    exceptions::throw_from_rcl_error(ret, "Failed to add time jump callback");
+  // Lookup the per-object mutex
+  g_clock_map_mutex.lock_shared();
+  std::mutex & clock_mutex = g_clock_mutex_map.at(this);
```

Comment:
Scoped would be nice here, since if this threw then the mutex would remain locked. You could use a function or a local lambda to enclose the scope but return the reference.